response_operations_ui/views/messages.py

```diff
@@ -125,19 +126,39 @@ def view_conversation(thread_id):
                            selected_survey=refined_thread[0]['survey'],
                            page=page,
                            closed_at=closed_at,
-                           thread_data=thread_conversation)
+                           thread_data=thread_conversation,
+                           show_mark_unread=_can_mark_as_unread(latest_message))
+
+
+@messages_bp.route('/mark_unread/<message_id>', methods=['GET'])
+@login_required
+def mark_message_unread(message_id):
+
+    msg_from = request.args.get(get_parameter('from'), type=str, default="")
+    msg_to = request.args.get(get_parameter('to'), type=str, default="")
+
+    message_controllers.add_unread_label(message_id)
+
+    marked_unread_message = f"Message from {msg_from} to {msg_to} marked unread"
```

Comment:
Have you checked that UX are okay with this message as it's different to whats on the design? It's better giving more detail but just making sure